pty/src/lib.rs

```diff
@@ -187,6 +188,17 @@ impl Child for std::process::Child {
     }
 
     fn kill(&mut self) -> IoResult<()> {
+        #[cfg(unix)]
+        {
+            // On unix, we send the SIGHUP signal instead of trying to kill
```

Comment:
While sending HUP first makes sense, the semantics of this method is that if it is successful, then it is assumed that the process has been killed.
This is important because calling `.wait()` will block until the process exits; if that process is ignoring HUP then it can potentially wedge the terminal GUI thread:

https://github.com/wez/wezterm/blob/main/mux/src/localpane.rs#L509-L510

The suggestion I made about allowing a grace period before upgrading to a regular kill allows both a chance for the shell to handle SIGHUP and for wezterm to not be held ransom by processes that won't die when we need them to!

I can't think of a good reason for the process directly attached to the pty here to ignore SIGHUP.  *Handling* it makes sense, but ignoring only makes sense for eg: a child process that is effectively a daemon process (spawned via `nohup`, in which case, it won't ever be associated directly with the pty, because `nohup` spawns a child process that detaches from the pty).